jenkins/parser/jenkins-parser-job.py

```diff
@@ -0,0 +1,611 @@
+        json.dump(old_running_builds_object, rotation_file)
+
+    return current_running_builds, old_running_builds
+
+
+def get_new_running_builds(job_dir, last_processed_log):
+    """Get list of new running builds that have been started after the last processed log."""
+    return [
+        dir.name
+        for dir in os.scandir(job_dir)
+        if dir.name.isdigit()
+        and int(dir.name) > int(last_processed_log)
+        and os.path.isfile(
+            functools.reduce(os.path.join, [job_dir, dir.name, "build.xml"])
+        )
+        and not grep(
+            functools.reduce(os.path.join, [job_dir, dir.name, "build.xml"]), "<result>"
+        )
+    ]
+
+
+def check_and_trigger_action(build_to_retry, job_dir, job_to_retry, error_list_action):
+    """Check failed build logs and trigger the appropiate action if a known error is found."""
+    build_dir_path = os.path.join(job_dir, build_to_retry)
+    log_file_path = os.path.join(build_dir_path, "log")
+    envvars_file_path = os.path.join(build_dir_path, "injectedEnvVars.txt")
+
+    # TODO: Try not to load everything on memory
+    text_log = open(log_file_path, errors="ignore")
+    lines = text_log.readlines()
+    text_log.close()
+
+    job_url = (
+        os.environ.get("JENKINS_URL") + "job/" + job_to_retry + "/" + build_to_retry
+    )
+
+    print("Parsing build #" + build_to_retry + " (" + job_url + ") ...")
+
+    regex_flag = 0
+    for error_and_action in error_list:
+        regex_and_action = error_and_action.split(" - ")
+        regex = regex_and_action[0]
+        action = regex_and_action[1]
+        for line in reversed(lines):
+            if re.search(regex, line):
+                print(
+                    "... Found message "
+                    + regex
+                    + " in "
+                    + log_file_path
+                    + ". Taking action ..."
+                )
+                if action == "retryBuild":
+                    trigger_retry = (
+                        os.environ.get("JENKINS_CLI_CMD")
+                        + " build jenkins-test-retry -p JOB_TO_RETRY="
+                        + job_to_retry
+                        + " -p BUILD_TO_RETRY="
+                        + build_to_retry
+                    )
+                    print(trigger_retry)
+                    os.system(trigger_retry)
+                else:
+                    # Take action on the nodes
+                    node_name = (
+                        grep(envvars_file_path, "NODE_NAME=", True)
+                        .split("=")[1]
+                        .replace("\n", "")
+                    )
+                    job_url = (
+                        os.environ.get("JENKINS_URL")
+                        + "job/"
+                        + job_to_retry
+                        + "/"
+                        + build_to_retry
+                    )
+                    node_url = "https://cmssdt.cern.ch/jenkins/computer/" + node_name
```

Comment:
we still have hard-coded  `https://cmssdt.cern.ch/jenkins` here

Reply:
Ups, I somehow undid the changes of commit 5bbf521fd096f68e600fca57f214f5ccc09cf1bb